crates/iceberg/src/error.rs

```diff
@@ -44,6 +44,10 @@ pub enum ErrorKind {
     ///
     /// This error is returned when given iceberg feature is not supported.
     FeatureUnsupported,
+    /// Error during io operation.
+    ///
+    /// This error is returned when doing io operations.
+    IoError,
```

Comment:
`IoError` is too general that no action can take. There are two error kind we can add there at least:

- `TableNotFound`: Table is not exist.
- `TablePermissionDenial`: User doesn't have enough permission to operate on this table.

Other errors can be treated as `Unexpected` until we find users may need to handle them.

What do you think?

Reply:
As a `FileIO` module, I don't think we should introduce `TableNotFound`, or  `TablePermissionDenial ` error, it should introduce error for  file not found or there is no permission to operator on file/dir. But I do admit that converting all opendal error into `IoError` would be too general. In fact when dealing with io, there are two kinds of error:

1. Unrecoverable: permission denied, file not found, operation not supported, etc. These errors should be `Unexpected`.
2. Recoverable: rate limit reached, timeout error, network failure. In this case, we should report it to caller and let caller to decide what to do.

I'll refine this later if we reached an agreement that this the correct abstraction.

Reply:
Makes sense.